sourdough/sourdough.py

```diff
@@ -227,13 +227,16 @@ def readVirtualMachineTag(tagName):
         hostname = v.get('hostname')
         username = v.get('user')
         password = v.get('password')
-        si= SmartConnect(host=hostname, user=username, pwd=password, sslContext=secure)
-        searcher = si.content.searchIndex
-        vm = searcher.FindByIp(ip=vm_ip, vmSearch=True)
-        if vm:
-          f = si.content.customFieldsManager.field
-          for k, v in [(x.name, v.value) for x in f for v in vm.customValue if x.key == v.key]:
-            vmwareTags[vm_ip][k] = v
+        try:
+          si= SmartConnect(host=hostname, user=username, pwd=password, sslContext=secure)
+          searcher = si.content.searchIndex
+          vm = searcher.FindByIp(ip=vm_ip, vmSearch=True)
+          if vm:
+            f = si.content.customFieldsManager.field
+            for k, v in [(x.name, v.value) for x in f for v in vm.customValue if x.key == v.key]:
+              vmwareTags[vm_ip][k] = v
+        except:
```

Comment:
Is there any particular error you're trying to rescue from? Generally, it's better to rescue from particular exceptions than from all of them especially in critical paths where it is expected to succeed.

Reply:
My bad - updating to catch just `socket.error`